From a06bb9a8b91e7e47700a084d52a2624f9dadc3fe Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 7 Sep 2018 11:34:13 +0530
Subject: [PATCH 0183/1051] net: octeontx2: Start/stop traffic at NPC instead
 of at CGX

Since a single CGX LMAC interface will be shared across a PF
and it's VFs, enabling/disabling RxTx at CGX based on PF's
logical interface UP/DOWN changes will impact VFs.

So instead of using CGX, enable/disable corresponding NPC MCAM
entries to start/stop receiving traffic.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_pf.c   | 2 +-
 drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
index cbe01199879c..84711af50e34 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
@@ -725,7 +725,7 @@ int otx2_stop(struct net_device *netdev)
 	struct otx2_qset *qset = &pf->qset;
 	int qidx, vec;
 
-	/* First stop packet Rx/Tx at CGX */
+	/* First stop packet Rx/Tx */
 	otx2_rxtx_enable(pf, false);
 
 	/* Disable link notifications */
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
index c3ea30bee3e9..90fc85a2a7be 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
@@ -561,9 +561,9 @@ int otx2_rxtx_enable(struct otx2_nic *pfvf, bool enable)
 		return 0;
 
 	if (enable)
-		msg = otx2_mbox_alloc_msg_CGX_START_RXTX(&pfvf->mbox);
+		msg = otx2_mbox_alloc_msg_NIX_LF_START_RX(&pfvf->mbox);
 	else
-		msg = otx2_mbox_alloc_msg_CGX_STOP_RXTX(&pfvf->mbox);
+		msg = otx2_mbox_alloc_msg_NIX_LF_STOP_RX(&pfvf->mbox);
 
 	if (!msg)
 		return -ENOMEM;
-- 
2.17.1

